### PR TITLE
metrics extended

### PR DESCRIPTION
**Pull request by synarete**
Provide extra-fine metrics information. In particular,. parse the `open_files` section in `smbstatus --json` and provide more metrics on open files.

### CODE REVIEW

**Review by anoopcs9 — COMMENTED** [review]
See below for a small query.

File: internal/metrics/smbstatus.go [hunk 1]
```diff
@@ -70,10 +70,59 @@ type SMBStatusFileID struct {
...
+}
+
+// SMBStatusOpenInfo represents a single entry open_files/opens output field
+type SMBStatusOpenInfo struct {
+	UID         int                     `json:"uid"`
+	ShareFileID string                  `json:"share_file_id"`
+	OpenedAt    string                  `json:"opened_at"`
+	ShareMode   SMBStatusOpenShareMode  `json:"sharemode"`
+	AccessMask  SMBStatusOpenAccessMask `json:"access_mask"`
+	OpLock      SMBStatusOpenOpLock     `json:"oplock"`
```
[the remark this answers is not in the document]
Comment (synarete) [reply internal/metrics/smbstatus.go:87]: Yes, absolutely. It is part of my `TODO` list, but for now I want a minimal viable set of metrics. Do you think `leases` should be part of it or may I defer it to next cycle?

[the remark this answers is not in the document]
Comment (anoopcs9) [reply internal/metrics/smbstatus.go:87]: `leases` can be considered as an upgrade to `oplocks`. I could see `oplocks`(but not yet used) and thus the question. You may choose to drop `oplocks` or add `leases` alongside.

[the remark this answers is not in the document]
Comment (synarete) [reply internal/metrics/smbstatus.go:87]: @anoopcs9 Added `Lease` entry (after `OpLock`). Please re-review.

File: internal/metrics/smbstatus_test.go [hunk 2]
```diff
@@ -602,3 +616,16 @@ func TestParseSMBStatusLocks(t *testing.T) {
 	assert.Equal(t, lock2.FileID.Inode, int64(52))
 	assert.Equal(t, lock2.NumPendingDeletes, 2)
 }
+
+func TestParseSMBStatusLease(t *testing.T) {
```
Comment (anoopcs9) [inline internal/metrics/smbstatus_test.go:620]: May be add a test for `oplocks` too?

**Review by anoopcs9 — APPROVED** [review]
lgtm, thanks.

File: internal/metrics/smbstatus_test.go (hunk 2 above)
Comment (synarete) [reply internal/metrics/smbstatus_test.go:620]: You are right -- I should have done so in the first place..

Comment (synarete) [reply internal/metrics/smbstatus_test.go]: Done

File: internal/metrics/smbstatus_test.go [hunk 3]
```diff
@@ -602,3 +608,28 @@ func TestParseSMBStatusLocks(t *testing.T) {
...
+				assert.Equal(t, lease.Handle, false)
+				assert.Equal(t, lease.Read, true)
+				assert.Equal(t, lease.Write, true)
+				assert.Equal(t, lease.Text, "RW")
+			} else {
+				assert.Equal(t, oplock.Exclusive, false)
+				assert.Equal(t, lease.Handle, false)
+				assert.Equal(t, lease.Read, false)
+				assert.Equal(t, lease.Write, false)
+				assert.Equal(t, lease.Text, "")
```
Comment (anoopcs9) [inline internal/metrics/smbstatus_test.go:631]: To cover other `SMBStatusOpenOpLock` types:
```
assert.Equal(t, oplock.Batch, true)
assert.Equal(t, oplock.LevelII, false)
assert.Equal(t, oplock.Text, "BATCH")
```

**Review by anoopcs9 — COMMENTED** [review]
See above for a small nit.

File: internal/metrics/smbstatus_test.go (hunk 3 above)
Comment (synarete) [reply internal/metrics/smbstatus_test.go:634]: Done

Comment (anoopcs9) [reply internal/metrics/smbstatus_test.go:634]: If you meant to include the additional checks I don't see it in the updated version.

**Pull request merged**